plutus-core/src/Language/PlutusCore/Universe/Core.hs

```diff
@@ -81,6 +83,7 @@ data Some f = forall a. Some (f a)
 
 -- | A particular type from a universe.
 newtype TypeIn uni a = TypeIn (uni a)
+    deriving (Data)
```

Comment:
Why do you need that? We don't have any derived instances there, because they generally need to be derived in a very particular way to make sense, see below in this file. So either remove this line or explain in a comment why `Data` is exceptional (I doubt it is, but who knows).